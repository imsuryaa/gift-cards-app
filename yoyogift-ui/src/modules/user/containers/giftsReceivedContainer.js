import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchReceivedCards, redeemCard } from './../state/actions/index';
import GiftsReceived from '../components/giftsReceived';
import CircularProgress from '@material-ui/core/CircularProgress';
import { Redirect } from 'react-router-dom';

export class GiftsReceivedContainer extends Component {

  componentDidMount() {
      if(this.props.user)
        this.props.fetchReceivedCards(this.props.user.email);
  }
  componentDidCatch(error, info) {
    console.log(error)
  }
  handleRedeemCard = (row) => {
    const rowId = row.id;
    const addObj = {
      id: row.id,
      senderEmail: row.senderEmail,
      receiverEmail: row.receiverEmail,
      cardName: row.cardName,
      cardPoints: row.cardPoints,
      cardShortDesc: row.cardShortDesc,
      cardImage: row.cardImage,
      cardIssueDate: row.cardIssueDate,
      cardExpiryDate: row.cardExpiryDate,
      isRedeemed: true
    }
    this.props.redeemCard(rowId, addObj);
  }

  render() {
    if (this.props.isLoggedIn) {
      if(!(this.props.receivedCards) || (this.props.receivedCards.length < 0)) {
        return <CircularProgress style={{marginLeft: '50%', marginTop: '10%'}} />            
      } else if (this.props.receivedCards.length === 0) {
        return <h2 style={{
          height: '40px',
          background: '#b3d9f7',
          color: 'white',
          textAlign: 'center',
          verticalAlign: 'middle',
          paddingTop: '5px',
          fontSize: '20px',
          fontWeight: '500'
        }}>
        NO DATA
        </h2>
        } else{
        return (
        <div>
          <GiftsReceived data={this.props.receivedCards} redeemCard={this.handleRedeemCard} />
        </div>
      )
    } 
  } else {
      return <Redirect to="/" />
  }   
  }
}

const mapStateToProps = (state) => {
    return {
        isLoggedIn: state.login.loginStatus,
        receivedCards: state.users.cards,
        user:state.login.detailsObject
    }
}

export default connect(mapStateToProps, {fetchReceivedCards, redeemCard})(GiftsReceivedContainer);
