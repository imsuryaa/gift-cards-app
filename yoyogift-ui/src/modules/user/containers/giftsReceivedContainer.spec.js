import React from "react";
import { mount, shallow } from "enzyme";

import { GiftsReceivedContainer } from "./giftsReceivedContainer";

describe("Gifts Received Test Suite", () => {
  it('should show circular progress', () => {
    const wrapper = mount(<GiftsReceivedContainer 
    isLoggedIn = {true}
    receivedCards={undefined}
    fetchReceivedCards={jest.fn()}
    />);

    expect(wrapper.find('div').length).toBe(1);
  });

  it('should show header with information', () => {
    const wrapper = mount(<GiftsReceivedContainer 
    isLoggedIn = {true}
    receivedCards={[]}
    fetchReceivedCards={jest.fn()}
    />);

    expect(wrapper.find('h2').length).toBe(1);
    expect(wrapper.find('h2').text()).toBe("NO DATA");
  });

  it('should show table with header and data', () => {
    //Mocks
    const data = [
      {
        cardName: 'A',
        cardPoints: 1,
        senderEmail: 'jalagamsurya.loriini@gmail.com',
        cardIssueDate: "Sun May 19 2019 15:43:25 GMT+0530 (India Standard Time)",
        cardExpiryDate: "Sun May 19 2019 15:43:25 GMT+0530 (India Standard Time)",
        isRedeemed: true
      },
      {
        cardName: 'B',
        cardPoints: 2,
        senderEmail: 'jalagamsurya.loriini@gmail.com',
        cardIssueDate: "Sun May 19 2019 15:43:25 GMT+0530 (India Standard Time)",
        cardExpiryDate: "Sun May 19 2019 15:43:25 GMT+0530 (India Standard Time)",
        isRedeemed: false
      },
      {
        cardName: 'C',
        cardPoints: 3,
        senderEmail: 'jalagamsurya.loriini@gmail.com',
        cardIssueDate: "Sun May 19 2019 15:43:25 GMT+0530 (India Standard Time)",
        cardExpiryDate: "Sun May 19 2019 15:43:25 GMT+0530 (India Standard Time)",
        isRedeemed: false
      }
    ];
    const wrapper = mount(<GiftsReceivedContainer 
    isLoggedIn = {true}
    receivedCards={data}
    fetchReceivedCards={jest.fn()}
    />);
    
    expect(wrapper.find('tr').length).toBe(4);
    expect(wrapper.find('th').length).toBe(10);
    expect(wrapper.find('td').length).toBe(18);
    expect(wrapper.find('button').length).toBe(2);
  });

  it('should call handle redeem card function', () => {
    //Mocks
    const data = [
      {
        cardName: 'A',
        cardPoints: 1,
        senderEmail: 'jalagamsurya.loriini@gmail.com',
        cardIssueDate: "Sun May 19 2019 15:43:25 GMT+0530 (India Standard Time)",
        cardExpiryDate: "Sun May 19 2019 15:43:25 GMT+0530 (India Standard Time)",
        isRedeemed: true
      },
      {
        cardName: 'B',
        cardPoints: 2,
        senderEmail: 'jalagamsurya.loriini@gmail.com',
        cardIssueDate: "Sun May 19 2019 15:43:25 GMT+0530 (India Standard Time)",
        cardExpiryDate: "Sun May 19 2019 15:43:25 GMT+0530 (India Standard Time)",
        isRedeemed: false
      },
      {
        cardName: 'C',
        cardPoints: 3,
        senderEmail: 'jalagamsurya.loriini@gmail.com',
        cardIssueDate: "Sun May 19 2019 15:43:25 GMT+0530 (India Standard Time)",
        cardExpiryDate: "Sun May 19 2019 15:43:25 GMT+0530 (India Standard Time)",
        isRedeemed: false
      }
    ];

    const wrapper = shallow(<GiftsReceivedContainer 
    isLoggedIn = {true}
    receivedCards={data}
    fetchReceivedCards={jest.fn()}
    redeemCard={jest.fn()}
    />);
    const component = wrapper.instance();
    const spy =  jest.spyOn(component, 'handleRedeemCard');

    component.handleRedeemCard(data[1]);
    component.handleRedeemCard(data[2]);
   
   expect(component.handleRedeemCard).toHaveBeenCalled();
   expect(component.handleRedeemCard).toHaveBeenCalledTimes(2)
  });
  
  it('should render the expected HTML', () => {
    expect(mount(<GiftsReceivedContainer 
    isLoggedIn = {true}
    receivedCards={undefined}
    fetchReceivedCards={jest.fn()}
    />
    ).html()).toMatchSnapshot();
  });

  it('should render the expected HTML', () => {
    expect(mount(<GiftsReceivedContainer 
    isLoggedIn = {true}
    receivedCards={[]}
    fetchReceivedCards={jest.fn()}
    />
    ).html()).toMatchSnapshot();
  });

  it('should render the expected HTML', () => {
    //Mocks
    const data = [
      {
        cardName: 'A',
        cardPoints: 1,
        senderEmail: 'jalagamsurya.loriini@gmail.com',
        cardIssueDate: "Sun May 19 2019 15:43:25 GMT+0530 (India Standard Time)",
        cardExpiryDate: "Sun May 19 2019 15:43:25 GMT+0530 (India Standard Time)",
        isRedeemed: true
      },
      {
        cardName: 'B',
        cardPoints: 2,
        senderEmail: 'jalagamsurya.loriini@gmail.com',
        cardIssueDate: "Sun May 19 2019 15:43:25 GMT+0530 (India Standard Time)",
        cardExpiryDate: "Sun May 19 2019 15:43:25 GMT+0530 (India Standard Time)",
        isRedeemed: false
      },
      {
        cardName: 'C',
        cardPoints: 3,
        senderEmail: 'jalagamsurya.loriini@gmail.com',
        cardIssueDate: "Sun May 19 2019 15:43:25 GMT+0530 (India Standard Time)",
        cardExpiryDate: "Sun May 19 2019 15:43:25 GMT+0530 (India Standard Time)",
        isRedeemed: false
      }
    ];
    expect(mount(<GiftsReceivedContainer 
    isLoggedIn = {true}
    receivedCards={data}
    fetchReceivedCards={jest.fn()}
    />
    ).html()).toMatchSnapshot();
  });

});
