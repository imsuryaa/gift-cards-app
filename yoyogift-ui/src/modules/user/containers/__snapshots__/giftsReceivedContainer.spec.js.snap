// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gifts Received Test Suite should render the expected HTML 1`] = `"<div class=\\"MuiCircularProgress-root-1 MuiCircularProgress-colorPrimary-4 MuiCircularProgress-indeterminate-3\\" style=\\"width: 40px; height: 40px; margin-left: 50%; margin-top: 10%;\\" role=\\"progressbar\\"><svg class=\\"MuiCircularProgress-svg-6\\" viewBox=\\"22 22 44 44\\"><circle class=\\"MuiCircularProgress-circle-7 MuiCircularProgress-circleIndeterminate-9\\" cx=\\"44\\" cy=\\"44\\" r=\\"20.2\\" fill=\\"none\\" stroke-width=\\"3.6\\"></circle></svg></div>"`;

exports[`Gifts Received Test Suite should render the expected HTML 2`] = `"<h2 style=\\"height: 40px; background: rgb(179, 217, 247); color: white; text-align: center; vertical-align: middle; padding-top: 5px; font-size: 20px; font-weight: 500;\\">NO DATA</h2>"`;

exports[`Gifts Received Test Suite should render the expected HTML 3`] = `"<div><div class=\\"MuiPaper-root-13 MuiPaper-elevation2-17 MuiPaper-rounded-14 GiftsReceived-root-11\\"><table class=\\"MuiTable-root-40 GiftsReceived-table-12\\"><thead class=\\"MuiTableHead-root-41\\"><tr class=\\"MuiTableRow-root-42 MuiTableRow-head-45\\"><th class=\\"MuiTableCell-root-47 MuiTableCell-head-48\\" scope=\\"col\\">CARD NAME</th><th class=\\"MuiTableCell-root-47 MuiTableCell-head-48\\" scope=\\"col\\">POINTS</th><th class=\\"MuiTableCell-root-47 MuiTableCell-head-48\\" scope=\\"col\\">RECEIVED FROM</th><th class=\\"MuiTableCell-root-47 MuiTableCell-head-48\\" scope=\\"col\\">CARD DESC</th><th class=\\"MuiTableCell-root-47 MuiTableCell-head-48\\" scope=\\"col\\">ISSUE DATE</th><th class=\\"MuiTableCell-root-47 MuiTableCell-head-48\\" scope=\\"col\\">EXPIRY DATE</th><th class=\\"MuiTableCell-root-47 MuiTableCell-head-48\\" scope=\\"col\\">REDEEM</th></tr></thead><tbody class=\\"MuiTableBody-root-59\\"><tr class=\\"MuiTableRow-root-42\\"><th class=\\"MuiTableCell-root-47 MuiTableCell-body-49\\" scope=\\"row\\">A</th><td class=\\"MuiTableCell-root-47 MuiTableCell-body-49\\">1</td><td class=\\"MuiTableCell-root-47 MuiTableCell-body-49\\">jalagamsurya.loriini@gmail.com</td><td class=\\"MuiTableCell-root-47 MuiTableCell-body-49\\"></td><td class=\\"MuiTableCell-root-47 MuiTableCell-body-49\\">2019-05-19</td><td class=\\"MuiTableCell-root-47 MuiTableCell-body-49\\">2019-05-19</td><td class=\\"MuiTableCell-root-47 MuiTableCell-body-49\\">Redeemed</td></tr><tr class=\\"MuiTableRow-root-42\\"><th class=\\"MuiTableCell-root-47 MuiTableCell-body-49\\" scope=\\"row\\">B</th><td class=\\"MuiTableCell-root-47 MuiTableCell-body-49\\">2</td><td class=\\"MuiTableCell-root-47 MuiTableCell-body-49\\">jalagamsurya.loriini@gmail.com</td><td class=\\"MuiTableCell-root-47 MuiTableCell-body-49\\"></td><td class=\\"MuiTableCell-root-47 MuiTableCell-body-49\\">2019-05-19</td><td class=\\"MuiTableCell-root-47 MuiTableCell-body-49\\">2019-05-19</td><td class=\\"MuiTableCell-root-47 MuiTableCell-body-49\\"><button class=\\"MuiButtonBase-root-86 MuiButton-root-60 MuiButton-contained-71 MuiButton-containedPrimary-72 MuiButton-raised-74 MuiButton-raisedPrimary-75\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label-61\\">Redeem</span><span class=\\"MuiTouchRipple-root-89\\"></span></button></td></tr><tr class=\\"MuiTableRow-root-42\\"><th class=\\"MuiTableCell-root-47 MuiTableCell-body-49\\" scope=\\"row\\">C</th><td class=\\"MuiTableCell-root-47 MuiTableCell-body-49\\">3</td><td class=\\"MuiTableCell-root-47 MuiTableCell-body-49\\">jalagamsurya.loriini@gmail.com</td><td class=\\"MuiTableCell-root-47 MuiTableCell-body-49\\"></td><td class=\\"MuiTableCell-root-47 MuiTableCell-body-49\\">2019-05-19</td><td class=\\"MuiTableCell-root-47 MuiTableCell-body-49\\">2019-05-19</td><td class=\\"MuiTableCell-root-47 MuiTableCell-body-49\\"><button class=\\"MuiButtonBase-root-86 MuiButton-root-60 MuiButton-contained-71 MuiButton-containedPrimary-72 MuiButton-raised-74 MuiButton-raisedPrimary-75\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label-61\\">Redeem</span><span class=\\"MuiTouchRipple-root-89\\"></span></button></td></tr></tbody></table></div></div>"`;
